### ISSUE TITLE
Provide PKGBUILD for Arch Linux

### ISSUE THREAD

**Issue by Pandiora** [roc-project/roc#334]
Since I encountered the same issues on Arch like described in [this issue](https://github.com/roc-project/roc/issues/266) (add pthread flag) and furthermore needed to reinstall python on a second machine running Arch too (missing scons module [add dependency?]), I wanted to ask if it would be possible to create a PKGBUILD which handles the compilation for us? Obviously lots of users would profit by just running yay or whatever package manager they're using and this project would be noticed by a wider audience.

I'm unexperienced when it comes to compilation and not sure if I could maintain such PKGBUILD. I already [looked into it](https://wiki.archlinux.org/index.php/PKGBUILD) and at how to create such file, but I probably couldn't fix compilation errors other users could encounter.

Just for reference your [Arch build instructions](https://roc-project.github.io/roc/docs/building/user_cookbook.html#arch-linux).

**Comment by gavv** [comment]
Hi, sorry for late reply!

Creating PKGBUILD (as well as packaging for other distros) would be appreciated. For that we'd need someone who uses Roc on Arch and wishes to maintain the PKGBUILD.

Personally I don't use Arch, but if you or someone else would like to become a package maintainer, I'd be happy to help in any way.

> I encountered the same issues on Arch like described in this issue (add pthread flag)

Hm, last time I wasn't able to reproduce it. I'll try once more in archlinux docker container.

> and furthermore needed to reinstall python on a second machine running Arch too (missing scons module [add dependency?])

Could you elaborate? What error did you encounter and what dependency are you talking about?

**Comment by gavv** [comment]
> Hm, last time I wasn't able to reproduce it. I'll try once more in archlinux docker container.

No, I can't reproduce the issue. Could you post your build commands and logs when the error ocurres?

**Comment by gavv** [comment]
> need to debug what is the cause and maybe open a new issue

Would be great!

> Depending the build errors with missing pthread flags I got this cmd error log and this build.log.

Thanks, I'll take a look into it.

**Comment by Pandiora** [comment]
@gavv I already created a PKGBUILD with a patch for pthread flag, but I still need to rewrite it so it just downloads the latest release.

Maybe you can help me: Your build instructions download pulseaudio, but with PKGBUILD's I'm just putting pulseaudio as dependency and for most users pulseaudio shouldn't get downloaded (again) this way. So my problem now is that in your docs you provide a parameter `--with-pulseaudio=<path>`, but I don't know how to reference the path (`/usr/lib`) correctly, since the building environment is part of a fakeroot. I already asked some people on Arch IRC Channels, but they couldn't provide me a solution either.

**Comment by gavv** [comment]
Hi, sorry for late reply, I was too busy and tired last month.

**Comment by gavv** [comment]
Regarding `-pthread` error. Thanks for detailed clarifications.

It seems that `-pthread` option is handled by `libtool` and it decides whether it is needed / supported on the current platform. It seems that on some platforms, it fails to add it. Other people report similar problems too: https://github.com/kevinkreiser/prime_server/issues/41

I've pushed a patch to `develop` (fd31def45268b18b6d24fe68bc84357c36d816a4) that adds `-pthread` when we're using a GNU-compatible toolchain. Could you please check whether it fixes the problem for you?

**Comment by gavv** [comment]
Regarding PulseAudio.

Unfortunately, PulseAudio doesn't provide an official API and infrastructure for external (out-of-tree) modules. So nominally it is extensible by modules, but all modules have to be part of the PulseAudio repository.

People tend to create out-of-tree modules from time to time (for their own risk), but building such modules would require fully built PulseAudio source tree. The `config.h` and other paths you mentioned are part of the PulseAudio source tree, but they're not installed on the system after building the sources.

When you're using `--build-3rdparty=pulseaudio`, Roc builds and uses the full PulseAudio source tree. When you're using `--with-pulseaudio=DIR`, you should download and build it by yourself.

Does it sucks? Yes. The proper solution would be to move our PulseAudio modules to upstream. I've plans to do it and even discussed it with upstream already, but there is still quite a lot of work to be done. As usual, any help is appreciated. If no one helps earlier, I think I'll look into it after next 4-5 minor releases of Roc, after finishing more important features.

Small note: if you're OK with building PulseAudio by Roc, but just want to prevent Roc from *downloading* anything, you can put the PulseAudio tarball into the `vendor` directory and `--build-3rdparty` will automatically use it. That directory is [a sub-module](https://github.com/roc-project/vendor) and it already contains some tarballs.

**Comment by Pandiora** [comment]
@gavv Thanks a lot, that clarified a lot of my questions. I still didn't uploaded a PKGBUILD since I was busy myself and unsure if I should create a dev-version or a release-based version. For now I think it would be more appropriate to go for releases only and patch the pthread part for the latest release until your implementation is part of a new release. I will report back when I finished and tested the PKGBUILD.

**Comment by gavv** [comment]
I'm planning to release 0.1.5 in a few days, which will include the pthread fix.

**Comment by Pandiora** [comment]
Okay, it took me so long now, I can wait until then. :)

**Comment by gavv** [comment]
Meanwhile, could you please check that the fresh develop fixes the pthread issue for you? (without patching 3rdparty.py)

**Comment by Pandiora** [comment]
@gavv I'm still getting errors: https://gist.github.com/Pandiora/70b2e52385b5112bd5edd816318fbe95

**Comment by gavv** [comment]
Strange, the current code should add either `-pthread` or `-lpthread`, depending on whether we're using a GNU toolchain. And in your log there is neither of them.

What commit are you using? Did you run something like:

```
git fetch origin
git checkout develop
git reset --hard origin/develop
```

or did a fresh clone?

**Comment by Pandiora** [comment]
@gavv Okay, sorry, my mistake. It were a fresh clone, but I tried to use the cpputest from AUR, which didn't worked like I expected. It now seems to add the flags correctly.

**Comment by gavv** [comment]
I've tagged the 0.1.5 release: https://github.com/roc-project/roc/releases/tag/v0.1.5

**Comment by Pandiora** [comment]
Alright, thank you, I will alter and add the PKGBUILD when I've some free time.

**Comment by brorbw** [comment]
Is there an update on this?

**Comment by Pandiora** [comment]
Well, sorry for the late reply, didn't looked at Github for quite some time. I almost finished a script for the package and did a lot of testing, but gave up in between, since Arch-Chat is helpful in a way, I can't do or do not understand what they're asking for. No promise, but I try to look into it again, I still got the install-script lying around on my ssd.

**Comment by brorbw** [comment]
@Pandiora i would not mind helping

**Comment by dvzrv** [comment]
In case this project is still active, I'd be interested to package it for the official repositories on Arch Linux, given that we could extend pipewire's functionality with it.

**Comment by brorbw** [comment]
I stopped using arch a while back, but I still wouldn't mind helping

**Comment by gavv** [comment]
Hi, please let me know what could be done from my side.

**Comment by gavv** [comment]
Moving this to discussions.
